### PR TITLE
Make Role and Permission strings to allow for service-specific values

### PR DESCRIPTION
**Pull request by ajkannan**

### CODE REVIEW

File: gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/Policy.java [hunk 1]
```diff
@@ -42,120 +40,63 @@
...
-public class Policy extends IamPolicy<Policy.Role> {
+public class Policy extends IamPolicy<String> {
 
   private static final long serialVersionUID = -5573557282693961850L;
 
   /**
-   * Represents legacy roles in an IAM Policy.
+   * The project-level roles in an IAM policy. This enum is not an exhaustive list of all roles
+   * you can use in an IAM policy. You can also use service-specific roles (e.g.
+   * <i>roles/pubsub.editor</i>).  See the <i>Supported Cloud Platform Services</i> page for links
```
Comment (aozarov) [inline gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/Policy.java]: Do you think we should make it clear that it is a string - "roles/pubsub.editor"?

File: gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/Policy.java [hunk 2]
```diff
@@ -42,120 +40,63 @@
...
-      return new Role("roles/viewer", Type.VIEWER);
-    }
+    OWNER("roles/owner");
 
-    /**
-     * Returns a {@code Role} of type {@link Type#EDITOR EDITOR}.
-     */
-    public static Role editor() {
-      return new Role("roles/editor", Type.EDITOR);
+    String value;
```
Comment (aozarov) [inline gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/Policy.java]: `private final`  ?

File: gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/ResourceManager.java [hunk 3]
```diff
@@ -366,15 +336,20 @@ public String value() {
    * @see <a href=
    *     "https://cloud.google.com/resource-manager/reference/rest/v1beta1/projects/testIamPermissions">
    *     Resource Manager testIamPermissions</a>
+   * @see <a href=
+   *     "https://cloud.google.com/iam/#supported_cloud_platform_services">Supported Cloud Platform
+   *     Services</a>
    */
-  List<Boolean> testPermissions(String projectId, List<Permission> permissions);
+  List<Boolean> testPermissions(String projectId, List<String> permissions);
```
Comment (aozarov) [inline gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/ResourceManager.java]: per our earlier discussion. I am fine with having just one flavour of this method.

**Comment by aozarov** [comment]
Looks good. Feel free to merge after considering the comments above.

**Comment by ajkannan** [comment]
Incorporated all suggestions.

**Pull request merged**